js.go

```diff
@@ -44,10 +45,29 @@ type JetStream interface {
 type JetStreamManager interface {
 	// Create a stream.
 	AddStream(cfg *StreamConfig) (*StreamInfo, error)
-	// Create a consumer.
-	AddConsumer(stream string, cfg *ConsumerConfig) (*ConsumerInfo, error)
+	// Update a stream.
+	UpdateStream(cfg *StreamConfig) (*StreamInfo, error)
+	// Delete a stream.
+	DeleteStream(name string) error
 	// Stream information.
 	StreamInfo(stream string) (*StreamInfo, error)
+	// Snapshot stream.
+	SnapshotStream(name string, cfg *StreamSnapshotConfig) (io.Reader, error)
+	// Restore stream.
+	RestoreStream(name string, snapshot io.Reader) error
+	// Purge stream messages.
+	PurgeStream(name string) error
+	// List streams.
+	ListStreams(offset int) ([]*StreamInfo, error)
+
+	// Create a consumer.
+	AddConsumer(stream string, cfg *ConsumerConfig) (*ConsumerInfo, error)
+	// Delete a consumer.
+	DeleteConsumer(stream, consumer string) error
+	// Consumer information.
+	ConsumerInfo(stream, durable string) (*ConsumerInfo, error)
```

Comment:
This also comes off a subscription. `sub.ConsumerInfo()` which is how I use it most.

Also does not have to be durable, so just `name`